torchvision/models/resnet_v2.py

```diff
@@ -0,0 +1,230 @@
+        self.bn1 = nn.BatchNorm2d(inplanes)
+        self.relu1 = nn.ReLU(inplace=True)
+        self.conv1 = conv3x3(inplanes, planes, stride)
+        self.bn2 = nn.BatchNorm2d(planes)
+        self.conv2 = conv3x3(planes, planes)
+        self.relu2 = nn.ReLU(inplace=True)  # just to make better graph
+        self.downsample = downsample
+        self.stride = stride
+
+    def forward(self, x):
+
+        out = self.bn1(x)
+        out = self.relu1(out)
+        residual = self.downsample(out) if self.downsample is not None else x
+        out = self.conv1(out)
+
+        out = self.bn2(out)
+        out = self.relu2(out)
+        out = self.conv2(out)
+
+        return out + residual
+
+
+class BottleneckV2(nn.Module):
+    r"""Bottleneck V2 from
+    `"Identity Mappings in Deep Residual Networks"<https://arxiv.org/abs/1603.05027>`_ paper.
+    This is used for ResNet V2 for 50, 101, 152 layers.
+
+    Args:
+        inplanes (int): number of input channels.
+        planes (int): number of output channels.
+        stride (int): stride size.
+        downsample (Module) optional downsample module to downsample the input.
+    """
+    expansion = 4
+
+    def __init__(self, inplanes, planes, stride=1, downsample=None):
+        super(BottleneckV2, self).__init__()
+        self.bn1 = nn.BatchNorm2d(inplanes)
+        self.conv1 = nn.Conv2d(inplanes, planes, kernel_size=1, bias=False)
+        self.relu1 = nn.ReLU(inplace=True)
+        self.bn2 = nn.BatchNorm2d(planes)
+        self.conv2 = conv3x3(planes, planes, stride)
+        self.relu2 = nn.ReLU(inplace=True)
+        self.bn3 = nn.BatchNorm2d(planes)
+        self.conv3 = nn.Conv2d(planes, planes * self.expansion, kernel_size=1, bias=False)
+        self.relu3 = nn.ReLU(inplace=True)
+        self.downsample = downsample
+
+    def forward(self, x):
+
+        out = self.bn1(x)
+        out = self.relu1(out)
+
+        residual = self.downsample(out) if self.downsample is not None else x
+        out = self.conv1(out)
+
+        out = self.bn2(out)
+        out = self.relu2(out)
+        out = self.conv2(out)
+
+        out = self.bn3(out)
+        out = self.relu3(out)
+        out = self.conv3(out)
+
+        return out + residual
+
+
+class ResNetV2(nn.Module):
+    r"""ResNet V2 model from
+    `"Identity Mappings in Deep Residual Networks"<https://arxiv.org/abs/1603.05027>`_ paper.
+    Args:
+        block (Module) : class for the residual block. Options are BasicBlockV1, BottleneckV1.
+        layers (list of int) : numbers of layers in each block
+        num_classes (int) :, default 1000, number of classification classes.
+    """
+    def __init__(self, block, layers, num_classes=1000):
+        super(ResNetV2, self).__init__()
+        assert block in (BottleneckV2, BasicBlockV2), "Argument block should be BottleneckV2 or BasicBlockV2"
+        self.inplanes = 64
+
+        self.bn0 = nn.BatchNorm2d(3, affine=False, track_running_stats=False)
```

Comment:
I assume this is a hack to handle preprocessing when you migrate an existing model to pytorch?
For the purpose of re-implementing a model in pytorch you probably should remove it.

Reply:
@ppwwyyxx actually this batchnorm is to be coherent with [mxnet.gluon one](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/model_zoo/vision/resnet.py#L308) but probably the reason is what you say. In the original torch [implementation](https://github.com/facebook/fb.resnet.torch/blob/master/models/preresnet.lua#L146) there is no BN. So, yes, probably I should remove it.